### ISSUE TITLE
pmap over `num_particles` in SVI

### ISSUE THREAD

**Issue by snehjp2** [pyro-ppl/numpyro#1645]
Hi, 

In `Trace_ELBO`, the `num_particles` argument allows one to effectively introduce a batch size in estimating the ELBO gradient if `num_particles > 1`. By default, it's vectorized over the `num_particles`. Is it possible to also distribute the batch dimension over devices (e.g. when running on multiple GPUs). My particular application is prone to jax OOM errors and would benefit from distribution over `jax.pmap`.

**Comment by fehiepsi** [comment]
If you got OOM, you can set vectorize particles to False. You can also use PositionalSharding like in [MCMC](https://num.pyro.ai/en/stable/mcmc.html?highlight=sharding#numpyro.infer.mcmc.MCMC) I guess.

If you want to pmap over particles, could you make a PR for it? I think we can just simply allow a **callable** `vectorize_particles` and call it [here](https://github.com/pyro-ppl/numpyro/blob/6e3f00774d7f79b53008c332bc9d2e68c8702f24/numpyro/infer/elbo.py#L171).
